src/Eloquent/Builder.php

```diff
@@ -92,7 +92,7 @@ public function insertGetId(array $values, $sequence = null)
             return $this->model->getKey();
         }
 
-        return parent::insertGetId($values, $sequence);
```

Comment:
By calling `parent::insertGetId`, the `insertGetId` will be executed from [here](https://github.com/laravel/framework/blob/d0a6080db78a7e8a3316ff9c41eb86aa0866742d/src/Illuminate/Database/Eloquent/Builder.php#L1967-L1969) on [`Illuminate\Database\Query\Builder`](https://github.com/laravel/framework/blob/d0a6080db78a7e8a3316ff9c41eb86aa0866742d/src/Illuminate/Database/Query/Builder.php#L3379-L3388) but we override this method in our [`MongoDB\Laravel\Query\Builder`](https://github.com/mongodb/laravel-mongodb/blob/698711ce63f8662e1579944ba47854fa0c5366e8/src/Query/Builder.php#L649-L664). So, I fix this problem by calling this method on our `MongoDB\Laravel\Query\Builder`.

Reply:
It seems to be fixed by [#2661](https://github.com/mongodb/laravel-mongodb/pull/2661/files#diff-5ea252bbf9d6b385257af07ce12346559d3d4e1727eef473c50b081e28e10230R40). Both solutions will work, yours removes a layer of magics but it is more fragile if the implementation changes in Laravel.